dmypy and mypy behave differently with same settings

Consider this example:

```
class Config:
    url = None

    def __init__(self) -> None:
        self.url = 'xxx'
```

`mypy` says there are no errors, but `dmypy` with the same settings finds one:

```
$ mypy example.py
Success: no issues found in 1 source file

$ dmypy run -- --follow-imports=skip example.py
Daemon started
example.py:3: error: Need type annotation for 'url'
Found 1 error in 1 file (checked 1 source file)

```

my package versions:

```

$ python --version
Python 3.6.5 :: Anaconda, Inc.
$ mypy --version
mypy 0.750
```

--- a/docs/source/command_line.rst
+++ b/docs/source/command_line.rst
@@ -484,21 +484,27 @@ of the above sections.
 
 .. option:: --local-partial-types
 
-    By default, mypy won't check partial types spanning module top level or class top level.
-    This flag changes the behavior to only allow partial types at local level. The most common
-    cases are variables initialized with ``None`` but without explicit ``Optional`` annotation,
-    therefore their types can not be fully inferred and specified. For example:
+    In mypy, the most common cases for partial types are variables initialized using ``None``,
+    but without explicit ``Optional`` annotations. By default, mypy won't check partial types
+    spanning module top level or class top level. This flag changes the behavior to only allow
+    partial types at local level, therefore it disallows inferring variable type for ``None``
+    from two assignments in different scopes. For example:
 
     .. code-block:: python
 
         from typing import Optional
 
-        a = None  # error
+        a = None  # Need type annotation here if using --local-partial-types
         b = None  # type: Optional[int]
 
         class Foo:
-            bar = None  # error
-            boo = None  # type: Optional[int]
+            bar = None  # Need type annotation here if using --local-partial-types
+            baz = None  # type: Optional[int]
+
+            def __init__(self) -> None
+                self.bar = 1
+
+        reveal_type(Foo().bar)  # Union[int, None] without --local-partial-types
 
     Note: this option is always implicitly enabled in mypy daemon and will become
     enabled by default for mypy in a future release.

--- a/docs/source/command_line.rst
+++ b/docs/source/command_line.rst
@@ -482,6 +482,23 @@ of the above sections.
            # 'items' now has type List[List[str]]
            ...
 
+.. option:: --local-partial-types
+
+    By default, mypy won't check partial types spanning module top level or class top level.
+    This flag changes the behavior to only allow partial types at local level. For example:
+
+    .. code-block:: python
+
+        from typing import Optional
+
+        a = None  # error
+        b = None  # type: Optional[str]
+
+        class Foo:
+            bar = None  # error
+            boo = None  # type: Optional[int]
+
+
 .. option:: --no-implicit-reexport
 
     By default, imported values to a module are treated as exported and mypy allows

--- a/docs/source/command_line.rst
+++ b/docs/source/command_line.rst
@@ -485,19 +485,28 @@ of the above sections.
 .. option:: --local-partial-types
 
     By default, mypy won't check partial types spanning module top level or class top level.
-    This flag changes the behavior to only allow partial types at local level. For example:
+    This flag changes the behavior to only allow partial types at local level. The most common
+    cases are variables initialized with values such as ``[]``, ``{}`` where the explicit type
+    arguments ``T`` for ``List[T]``, ``Dict[T]`` is missing, or ``None`` without explicit
+    ``Option`` annotation, therefore their types can not be fully inferred and specified. For example:
 
     .. code-block:: python
 
-        from typing import Optional
+        from typing import Optional, List, Dict
 
         a = None  # error
-        b = None  # type: Optional[str]
+        b = None  # type: Optional[int]
+        l1 = []   # error
+        l2 = []   # type: List[int]
 
         class Foo:
             bar = None  # error
             boo = None  # type: Optional[int]
+            d1 = {}  # error
+            d2 = {}  # type: Dict[int, float]
 
+    Note: this option is always implicitly enabled in mypy daemon and will become
+    enabled by default for mypy in a future release.
 
 .. option:: --no-implicit-reexport
 

--- a/docs/source/command_line.rst
+++ b/docs/source/command_line.rst
@@ -486,24 +486,19 @@ of the above sections.
 
     By default, mypy won't check partial types spanning module top level or class top level.
     This flag changes the behavior to only allow partial types at local level. The most common
-    cases are variables initialized with values such as ``[]``, ``{}`` where the explicit type
-    arguments ``T`` for ``List[T]``, ``Dict[T]`` is missing, or ``None`` without explicit
-    ``Option`` annotation, therefore their types can not be fully inferred and specified. For example:
+    cases are variables initialized with ``None`` but without explicit ``Optional`` annotation,
+    therefore their types can not be fully inferred and specified. For example:
 
     .. code-block:: python
 
-        from typing import Optional, List, Dict
+        from typing import Optional
 
         a = None  # error
         b = None  # type: Optional[int]
-        l1 = []   # error
-        l2 = []   # type: List[int]
 
         class Foo:
             bar = None  # error
             boo = None  # type: Optional[int]
-            d1 = {}  # error
-            d2 = {}  # type: Dict[int, float]
 
     Note: this option is always implicitly enabled in mypy daemon and will become
     enabled by default for mypy in a future release.

